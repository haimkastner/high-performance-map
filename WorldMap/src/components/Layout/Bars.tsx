import Box from '@material-ui/core/Box';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import React, { useEffect, useState } from 'react';
import { ScreenPosition } from '../../models/models';
import { subscribeHookState } from '../../services/stats-store';
import { LayoutItem } from './LayoutItem';
import { DownBar, MapButtons, NavBar, UpBar } from './LayoutItems';
import { MapButtonsMenu } from './LayoutItems/MapButtonsMenu';

/**
 * to customize design for ui components in the future.
 */
const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    bars: {
      // for future custom design
    },
  }),
);

/**
 * The bars layout,
 * The bars are returned with their unique css design by using @LayoutItem .
 * @returns all the bars above our map.
 */
export function Bars() {
  const classes = useStyles();
  const [mapButtonsMenuCenterDis, setMapButtonsMenuCenterDis] = useState('calc(50vh - 210px)');
  const [showMapButtonsMenu, setShowMapButtonsMenu] = useState(false);
  const [showNavBar, setShowNavBar] = useState(false);
  const [showPlatformMenu, setShowPlatformMenu] = useState(false);
  const [selectedPlatformPosition, setSelectedPlatformPosition] = useState<ScreenPosition>({ x: 0, y: 0 });

  useEffect(() => {
    subscribeHookState('mapButtonsMenuCenterDis', setMapButtonsMenuCenterDis);
    subscribeHookState('showMapButtonsMenu', setShowMapButtonsMenu);
    subscribeHookState('showNavBar', setShowNavBar);
    subscribeHookState('showPlatformMenu', setShowPlatformMenu);
    subscribeHookState('selectedPlatformPosition', setSelectedPlatformPosition);
  }, []);

  return (
    <Box className={classes.bars}>
      <LayoutItem opacity={0.5} color={'#016976'} height={'41px'} top={0} width={'100vw'} left={0}>
        <UpBar />
      </LayoutItem>
      <LayoutItem opacity={0.5} color={'#016976'} height={'419px'} width={'60px'} left={0} top={'calc(50vh - 210px)'}>
        <MapButtons />
      </LayoutItem>
      {showMapButtonsMenu ? (
        <LayoutItem
          opacity={0.5}
          color={'red'}
          height={'500px'}
          width={'400px'}
          left={60}
          top={`calc(50vh + ${mapButtonsMenuCenterDis}px)`}
        >
          <MapButtonsMenu></MapButtonsMenu>
        </LayoutItem>
      ) : (
        <Box />
      )}
      {showPlatformMenu ? (
        <LayoutItem
          opacity={0.5}
          color={'gray'}
          top={`calc(${selectedPlatformPosition.y}px - 100px)`}
          left={`calc(${selectedPlatformPosition.x}px - 100px)`}
          height={'200px'}
          width={'200px'}
        >
          <Box> This is the platform menu </Box>
        </LayoutItem>
      ) : (
        <Box />
      )}
      <LayoutItem
        opacity={0.5}
        // color={"transparent"}
        color={'#016976'}
        height={'419px'}
        width={'60px'}
        right={0}
        top={'calc(50vh - 210px)'}
      >
        <MapButtons />
      </LayoutItem>
      {showNavBar ? (
        <LayoutItem
          opacity={0.5}
          color={'transparent'}
          height={'66px'}
          width={'1000px'}
          right={'calc(50vw - 500px)'}
          top={90}
        >
          <NavBar />
        </LayoutItem>
      ) : (
        <Box />
      )}
      <LayoutItem opacity={0.5} color={'#016976'} height={'41'} bottom={0} width={'100vw'} left={0}>
        <DownBar />
      </LayoutItem>
    </Box>
  );
}
