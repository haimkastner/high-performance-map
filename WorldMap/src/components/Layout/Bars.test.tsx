import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { Bars } from './Bars';

describe('Bar Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Bars />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches the snapshot', () => {
    const tree = renderer.create(<Bars />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // tslint:disable-next-line: no-empty
  describe('Bars Component Logic', () => {});
});
