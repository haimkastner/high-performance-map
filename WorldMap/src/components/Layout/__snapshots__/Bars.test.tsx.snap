// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar Component matches the snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-177 makeStyles-bars-175"
>
  <div
    className="MuiBox-root MuiBox-root-181 makeStyles-layout-178"
    id="container"
    style={
      Object {
        "bottom": "",
        "height": "41px",
        "left": 0,
        "right": "",
        "top": 0,
        "width": "100vw",
      }
    }
  >
    <div
      className="MuiBox-root MuiBox-root-182 makeStyles-color-179"
      id="background"
      style={
        Object {
          "backgroundColor": "#016976",
          "opacity": 0.5,
        }
      }
    />
    <div
      className="MuiBox-root MuiBox-root-183 makeStyles-item-180"
      id="content-box"
      style={Object {}}
    >
      <div
        className="MuiBox-root MuiBox-root-184"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            I am up button
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="MuiBox-root MuiBox-root-217 makeStyles-layout-178"
    id="container"
    style={
      Object {
        "bottom": "",
        "height": "419px",
        "left": 0,
        "right": "",
        "top": "calc(50vh - 210px)",
        "width": "60px",
      }
    }
  >
    <div
      className="MuiBox-root MuiBox-root-218 makeStyles-color-179"
      id="background"
      style={
        Object {
          "backgroundColor": "#016976",
          "opacity": 0.5,
        }
      }
    />
    <div
      className="MuiBox-root MuiBox-root-219 makeStyles-item-180"
      id="content-box"
      style={Object {}}
    >
      <div
        className="MuiGrid-root makeStyles-root-220 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
      >
        <div
          className="MuiBox-root MuiBox-root-324"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <div
                className="MuiBox-root MuiBox-root-325"
              >
                <img
                  height="19"
                  src="sensors-move@2x.png"
                  width="19"
                />
              </div>
            </span>
          </button>
        </div>
        <div
          className="MuiBox-root MuiBox-root-326"
        >
          <img
            height={25}
            src="adding-copy-shape-3@2x.png"
            width={25}
          />
        </div>
        <div
          className="MuiBox-root MuiBox-root-327"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <div
                className="MuiBox-root MuiBox-root-328"
              >
                <img
                  height={25}
                  src="display-copy-shape-2@2x.png"
                  width={25}
                />
              </div>
            </span>
          </button>
        </div>
        <div
          className="MuiBox-root MuiBox-root-329"
        >
          <img
            height={25}
            src="movement-path-4333-shape-1@2x.png"
            width={25}
          />
        </div>
        <div
          className="MuiBox-root MuiBox-root-330"
        >
          <img
            height={25}
            src="side-menu-shape-copy-2@2x.png"
            width={25}
          />
        </div>
        <div
          className="MuiBox-root MuiBox-root-331"
        >
          <img
            height={25}
            src="side-menu-shape-copy-2@2x.png"
            width={25}
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiBox-root MuiBox-root-332"
  />
  <div
    className="MuiBox-root MuiBox-root-333 makeStyles-layout-178"
    id="container"
    style={
      Object {
        "bottom": "",
        "height": "419px",
        "left": "",
        "right": 0,
        "top": "calc(50vh - 210px)",
        "width": "60px",
      }
    }
  >
    <div
      className="MuiBox-root MuiBox-root-334 makeStyles-color-179"
      id="background"
      style={
        Object {
          "backgroundColor": "#016976",
          "opacity": 0.5,
        }
      }
    />
    <div
      className="MuiBox-root MuiBox-root-335 makeStyles-item-180"
      id="content-box"
      style={Object {}}
    >
      <div
        className="MuiGrid-root makeStyles-root-220 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
      >
        <div
          className="MuiBox-root MuiBox-root-336"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <div
                className="MuiBox-root MuiBox-root-337"
              >
                <img
                  height="19"
                  src="sensors-move@2x.png"
                  width="19"
                />
              </div>
            </span>
          </button>
        </div>
        <div
          className="MuiBox-root MuiBox-root-338"
        >
          <img
            height={25}
            src="adding-copy-shape-3@2x.png"
            width={25}
          />
        </div>
        <div
          className="MuiBox-root MuiBox-root-339"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <div
                className="MuiBox-root MuiBox-root-340"
              >
                <img
                  height={25}
                  src="display-copy-shape-2@2x.png"
                  width={25}
                />
              </div>
            </span>
          </button>
        </div>
        <div
          className="MuiBox-root MuiBox-root-341"
        >
          <img
            height={25}
            src="movement-path-4333-shape-1@2x.png"
            width={25}
          />
        </div>
        <div
          className="MuiBox-root MuiBox-root-342"
        >
          <img
            height={25}
            src="side-menu-shape-copy-2@2x.png"
            width={25}
          />
        </div>
        <div
          className="MuiBox-root MuiBox-root-343"
        >
          <img
            height={25}
            src="side-menu-shape-copy-2@2x.png"
            width={25}
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiBox-root MuiBox-root-344"
  />
  <div
    className="MuiBox-root MuiBox-root-345 makeStyles-layout-178"
    id="container"
    style={
      Object {
        "bottom": 0,
        "height": "41",
        "left": 0,
        "right": "",
        "top": "",
        "width": "100vw",
      }
    }
  >
    <div
      className="MuiBox-root MuiBox-root-346 makeStyles-color-179"
      id="background"
      style={
        Object {
          "backgroundColor": "#016976",
          "opacity": 0.5,
        }
      }
    />
    <div
      className="MuiBox-root MuiBox-root-347 makeStyles-item-180"
      id="content-box"
      style={Object {}}
    >
      <div
        className="MuiBox-root MuiBox-root-348"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            I am DownBar button
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
